plugins/updater/src/updater.rs

```diff
@@ -642,7 +644,12 @@ impl Update {
                             }
                         }
                     }
+                    // if we have not returned early we should restore the backup
+                    std::fs::rename(tmp_app_image, &self.extract_path)?;
 
+                    // The convention here is to return Ok(()) if the right installer
+                    // was not found in the package, even though this means the update
+                    // failed.
```

Comment:
We should probably make this an error instead, e.g. `return Err(Error::AppimageNotFoundInZip)`